dataset_merge.py

```diff
@@ -0,0 +1,123 @@
+import os
+import argparse
+from datasets import (
+    DatasetDict,
+    load_dataset, 
+    concatenate_datasets
+)
+
+from huggingface_hub import create_repo
+from huggingface_hub.utils import HfHubHTTPError
+from utils import update_args
+
+KEEP_COLUMNS = ["generator", "prompt_id", "prompt", "messages", "category"]
+
+def check_args(args):
+    if args.first_ds_id is None or \
+        args.first_ds_train_split is None or \
+        args.first_ds_test_split is None:
+        raise ValueError("some arguments for the first dataset are missing")
+
+    if args.second_ds_id is None or \
+        args.second_ds_train_split is None:
+        raise ValueError("some arguments for the second dataset are missing")
+
+    if args.result_ds_id is None or \
+        args.result_ds_train_split is None or \
+        args.result_ds_test_split is None:
+        raise ValueError("some arguments for the result dataset are missing")
+
+def merge_datasets(args):
+    try:
+        check_args(args)
+    except ValueError as e:
+        print(str(e))
+
+    first_train_ds = load_dataset(args.first_ds_id, split=args.first_ds_train_split)
+    first_test_ds = load_dataset(args.first_ds_id, split=args.first_ds_test_split)
+
+    second_train_ds = load_dataset(args.second_ds_id, split=args.second_ds_train_split)
+    result_train_ds = concatenate_datasets([first_train_ds, second_train_ds])
+    result_test_ds = first_test_ds
+
+    if args.second_ds_test_split:
+        second_test_ds = load_dataset(args.second_ds_id, split=args.second_ds_test_split)
+        result_test_ds = concatenate_datasets([first_test_ds, second_test_ds])
+
+    result_ds = DatasetDict(
+        {
+            args.result_ds_train_split: result_train_ds,
+            args.result_ds_test_split: result_test_ds
+        }
+    )
+
+    if args.push_result_ds_to_hf_hub and args.hf_token is not None:
```

Comment:
We shouldn't pass `hf_token` from the CLI actually. It's a bad practice. Because if someone ends up logging `args` (which is common), it will silently log their `hf_token` too. 

Instead, we should always recommend running `huggingface-cli login` or setting the `HUGGING_FACE_HUB_TOKEN` env variable. 

Sorry this slipped from my mind in the earlier reviews.